aws/eks-customer/scripts/deploy-utility.sh

```diff
@@ -0,0 +1,132 @@
+#/usr/bin/env bash
+
+set -e -x
+
+source $(dirname "$0")/utils.sh
+
+utilities_json=$1
+
+function add_cluster(){
+  echo "Adding cluster to cluster-values.yaml"
+  
+	cluster_exists=$(yq eval ".clusters[] | select(.name == \"${CLUSTER_NAME}\")" $cluster_yaml)
+
+    if [ -z "$cluster_exists" ]; then
+        yq eval ".clusters += [{
+            \"name\": \"${CLUSTER_NAME}\",
+            \"type\": \"eks\",
+            \"labels\": {
+                    \"cluster-type\": \"customer\",
+                    \"cluster-id\": \"${CLUSTER_NAME}\"
+            },
+            \"api_server\": \"${API_SERVER}\",
+            \"clusterName\": \"${CLUSTER_NAME}\",
+            \"argoCDRoleARN\": \"${ARGOCD_ROLE_ARN}\",
+            \"caData\": \"${CA_DATA}\"
+        }]" -i $cluster_yaml
+    else
+      # Update the cluster if it already exists
+      yq eval "(.clusters[] | select(.name == \"${CLUSTER_NAME}\") | .labels."cluster-type") = \"customer\"" -i $cluster_yaml
+      yq eval "(.clusters[] | select(.name == \"${CLUSTER_NAME}\") | .labels."cluster-id") = \"${CLUSTER_NAME}\"" -i $cluster_yaml
+      yq eval "(.clusters[] | select(.name == \"${CLUSTER_NAME}\") | .api_server) = \"${API_SERVER}\"" -i $cluster_yaml
+      yq eval "(.clusters[] | select(.name == \"${CLUSTER_NAME}\") | .argoCDRoleARN) = \"${ARGOCD_ROLE_ARN}\"" -i $cluster_yaml
+      yq eval "(.clusters[] | select(.name == \"${CLUSTER_NAME}\") | .caData) = \"${CA_DATA}\"" -i $cluster_yaml
+      echo "Cluster already exists, updating the cluster."
+    fi
+  
+  echo "Commiting changes: Adding cluster ${CLUSTER_NAME}"
+  stage_changes $cluster_yaml
+  commit_changes "Adding cluster: ${CLUSTER_NAME}" $cluster_yaml
+}
+
+function create_cluster_folder() {
+	echo "Creating custom values file"
+	mkdir $gitops_apps_dir/${ENV}/helm-values/${CLUSTER_NAME} || true
```

Comment:
`|| true` suppresses all errors, was there any reason for that?

Reply:
This is to avoid errors when the file already exists.